R/set_args.R

```diff
@@ -54,6 +54,23 @@ set_legend_args <- function(object, fun.name, ..., legend.name) {
   
   paramsAll <- set_args(fun.name, list(...))
   
+  if(length(legend.name) > 1){
+    paramsAll_df <- as.data.frame(paramsAll, stringsAsFactors = FALSE)
```

Comment:
Would it work to move this logic into get_legend_args?  It seems like it shouldn't be the responsibility of set_args to know quirks that exist with legends.  I think here we just need to be able to handle whatever get_legend_args hands back.  I'm not sure that is right though.